src/Testing/CodeHacking/CodeHackerTestHook.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * CodeHackerTestHook class file.
+ *
+ * @package WooCommerce/Testing
+ */
+
+// phpcs:disable Squiz.Commenting.FunctionComment.Missing, PHPCompatibility.FunctionDeclarations
+
+namespace Automattic\WooCommerce\Testing\CodeHacking;
+
+use PHPUnit\Runner\BeforeTestHook;
+use PHPUnit\Runner\AfterTestHook;
+use PHPUnit\Util\Test;
+use ReflectionClass;
+use ReflectionMethod;
+use Exception;
+
+/**
+ * Helper to use the CodeHacker class in PHPUnit.
+ *
+ * How to use:
+ *
+ * 1. Add this to phpunit.xml:
+*
+ *    <extensions>
+ *      <extension class="CodeHackerTestHook" />
+ *    </extensions>
+ *
+ * 2. Add the following to the test classes:
+ *
+ *    use Automattic\WooCommerce\Testing\CodeHacking\CodeHacker;
+ *    use Automattic\WooCommerce\Testing\CodeHacking\Hacks\...
+ *
+ *    public static function before_all($method_name) {
+ *      CodeHacker::add_hack(...);
+ *      //Register as many hacks as needed
+ *      CodeHacker::enable();
+ *    }
+ *
+ *    $method_name is optional, 'before_all()' is also a valid method signature.
+ *
+ *    You can also define a test-specific 'before_{$test_method_name}' hook.
+ *    If both exist, first 'before_all' will be executed, then the test-specific one.
+ *
+ * 3. Additionally, you can register hacks via class/method annotations
+ *    (note that then you don't need the `use`s anymore):
+ *
+ *    /**
+ *     * @hack HackClassName param1 param2
+ *     * /
+ *    class Some_Test
+ *    {
+ *       /**
+ *        * @hack HackClassName param1 param2
+ *        * /
+ *       public function test_something() {
+ *       }
+ *    }
+ *
+ *    If the class name ends with 'Hack' you can omit that suffix in the annotation (e.g. 'Foo' instead of 'FooHack').
+ *    Parameters specified after the class name will be passed to the class constructor.
+ *    Hacks defined as class annotations will be applied to all tests.
+ */
+final class CodeHackerTestHook implements BeforeTestHook, AfterTestHook {
+
+	public function executeAfterTest( string $test, float $time ): void {
+		if ( ! CodeHacker::has_persistent_hacks() ) {
+			CodeHacker::disable();
+		}
+	}
+
+	public function executeBeforeTest( string $test ): void {
+		/**
+		 * Possible formats of $test:
+		 * TestClass::TestMethod
+		 * TestClass::TestMethod with data set #...
+		 * Warning
+		 */
+		$parts = explode( '::', $test );
+		if ( count( $parts ) < 2 ) {
+			return;
```

Comment:
This is not an error. This check is in place because the hook is sometimes invoked with the word "Warning" instead of a class name, and these cases must be discarded.

Reply:
> the hook is sometimes invoked with the word "Warning" instead of a class name
just out of curiosity, when will this happen?